### PR TITLE
ReturnnConfig: parse delayed ops in prolog/epilog

### PR DESCRIPTION
**Pull request by vieting**
In one of my `ReturnConfig`'s prolog, I have a `DelayedFormat` object. Currently, this cannot be parsed, therefore I'd like to add it.

### CODE REVIEW

File: returnn/config.py [hunk 1]
```diff
@@ -227,6 +227,8 @@ def __parse_python(self, code, name=None):
             return ""
         if isinstance(code, str):
             return code
+        if isinstance(code, DelayedBase):
+            return code.get()
```
Comment (michelwi) [inline returnn/config.py:231]: I am not sure that `DelayedBase.get()` always returned a string. 

should we assert that here?
would we like to handle other types? Maybe a delayed function should be turned into a function and then handled as source code?

**Comment by vieting** [comment]
The import was lost when rebasing... I added it again.

File: returnn/config.py (hunk 1 above)
Comment (vieting) [reply returnn/config.py:232]: That't true. We could do
```suggestion
        if isinstance(code, DelayedBase):
            return self.__parse_python(code.get())
```

**Comment by JackTemaki** [comment]
The ReturnnConfig should already support this via `config = instanciate_delayed(config)` in the `_serialize` function. @vieting can you say for which case this does not work?

File: returnn/config.py (hunk 1 above)
Comment (JackTemaki) [reply returnn/config.py:232]: This suggestion seems like the most consistent way.

**Review by michelwi — APPROVED** [review]

**Comment by vieting** [comment]
@JackTemaki what do you think?

**Review by JackTemaki — APPROVED** [review]

**Comment by JackTemaki** [comment]
I just noticed that this functionality is actually not hash-safe...., because the parse op is also called during the hash computation itself. So the hash will be different depending on if the variable exists or not.

@michelwi  shame on us I guess for not noticing this...

**Pull request merged**